index.js

```diff
@@ -11,3 +10,52 @@ const mongoose = require('./lib/');
 module.exports = mongoose;
 module.exports.default = mongoose;
 module.exports.mongoose = mongoose;
+
+// Re-export for ESM support
+module.exports.cast = mongoose.cast;
+module.exports.STATES = mongoose.STATES;
+module.exports.setDriver = mongoose.setDriver;
+module.exports.set = mongoose.set;
+module.exports.get = mongoose.get;
+module.exports.createConnection = mongoose.createConnection;
+module.exports.connect = mongoose.connect;
+module.exports.disconnect = mongoose.disconnect;
+module.exports.startSession = mongoose.startSession;
+module.exports.pluralize = mongoose.pluralize;
+module.exports.model = mongoose.model;
+module.exports.deleteModel = mongoose.deleteModel;
+module.exports.modelNames = mongoose.modelNames;
+module.exports.plugin = mongoose.plugin;
+// module.exports.connection = mongoose.connection;
+module.exports.connections = mongoose.connections;
+// module.exports.Collection = mongoose.Collection;
+// module.exports.Connection = mongoose.Connection;
```

Comment:
imo add a comment in code about why these things aren't exported. So in the future, when anyone checks these codes, they'll be able to understand why these things aren't exported.

Reply:
Do we export these by our typings to directly be instantiated? If so, we should not directly export them.

Reply:
yeah. Typings exports them

Reply:
I second @imranbarbhuiya 
A comment specifying why those are not exported would be nice.

Otherwise, LGTM. 👍

Reply:
The weird quirk is that you can do `import { connection } from 'mongoose'` in TypeScript even without this change. This change is just for Node's particular ESM implementation.

I have an alternative idea: just use `Object.defineProperty()` to create a getter for these particular properties. That should work for Node's ESM implementation.

Reply:
Actually, looks like `defineProperty()` only works if you set `value`, not `get`:

```javascript
// Works
Object.defineProperty(module.exports, 'connection', {
  value: mongoose.connection
});

// Doesn't work
Object.defineProperty(module.exports, 'connection', {
  get: () => mongoose.connection
});
```

So it looks like this idea won't work.

2 options:

1. Leave `connection`, `Collection`, `Connection` commented out. You'll still be able to use them in TypeScript, both as named imports and as `const connection = mongoose.connection`. Just won't be able to use them as named imports with Node ESM.
2. Uncomment them. However, this is a potentially breaking change because any TypeScript code that uses a custom driver and `Connection` as a named import will (probably) no longer work.

I'm leaning towards (1) because that's the least risky approach. Will do some more tinkering and see if there are other considerations.

Reply:
i think you could also just:
- define it on another object (or its own value) and then export it
- use a [Proxy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) object (or similar)

though i did not try this, so these are just thoughts

Reply:
The problem is that Node ESM finds named imports by parsing the source, not by actually running it. So proxy wouldn't help. Defining `connection` etc. on another object would be better, but still a more significant change than I'd like.